src/readers/FeigReader.java

```diff
@@ -439,61 +272,44 @@ public boolean clearReader() {
 	}
 
 	@Override
-	public boolean connect() {
-		logger.info("Connecting to FEIG reader.");
-		
-		// Initialize FEIG Reader.
-		if (!initiateFeigReader()) {
-			logger.error("FEIG Reader: Error - CANNOT INITIALIZE");
-			running = false;
-			connected = false;
-			return false;
-		} else {
-			try {
-				openUSBPort();
-				logger.error("USB Connection: ESTABLISHED");
+	protected HashMap<String, BibTag> getTags() throws FedmException, FePortDriverException, FeReaderDriverException {
+		HashMap<String, BibTag> tags = new HashMap<String, BibTag>();
+
+		HashMap<String, FedmIscTagHandler> inventory = fedm.tagInventory(true, (byte) 0, (byte) 1);
+
+		FedmIscTagHandler th = null;
+		FedmIscTagHandler_Result res = new FedmIscTagHandler_Result();
+		int back = 0;
+		int tagDriver = 0;
+
+		// Read tags with data.
+		for (Map.Entry<String, FedmIscTagHandler> e : inventory.entrySet()) {
+			String uid = (String) e.getKey();
+			th = (FedmIscTagHandler) e.getValue();
+			th = fedm.tagSelect(th, tagDriver);
+
+			if (th instanceof FedmIscTagHandler_ISO15693) {
+				FedmIscTagHandler_ISO15693 thIso = (FedmIscTagHandler_ISO15693) th;
+				back = thIso.readMultipleBlocksWithSecStatus(0, 8, res);
 				
-				connected = true;
-			}
-			catch (Exception e) {
-				logger.error("USB Connection Error: " + e.getMessage());
-				running = false;
-				connected = false;
-				return false;
+				// 0 = success.
+				if (back == 0) {
```

Comment:
Should the variable "back" have been "block"? If not what do "back" mean in this context!